R/billingInfo.R

```diff
@@ -0,0 +1,137 @@
+#' @rdname getBillingInfo
+#' @export 
+#' @importFrom tibble as_tibble
+
+getBillingInfo <- function(asDataFrame = TRUE, userId) {
+  billingInfo <- getResource(paste0("/billingAccounts/", userId), task = "Getting billing account info")
+  if (asDataFrame == TRUE) {
+    billingInfo <- tibble::as_tibble(billingInfo)
+    return(billingInfo)
+  } else {
+    return(billingInfo)
+  }
+}
+
+
+#' @title getBillingDatabases
+#' @description Data for all databases under billing account
+#' @param asDataFrame Output as data.frame, Default: TRUE
+#' @param userId User ID
+#' @return Information on databases under billing account in list or data.frame output
+#' @rdname getBillingDatabases
+#' @export 
+#' @importFrom tibble tibble
+
+getBillingDatabases <- function(asDataFrame = TRUE, userId) {
+  billingDatabases <- getResource(paste0("/billingAccounts/", userId, "/databases"), 
+                                  task = "Getting billing account databases")
+  if (asDataFrame == TRUE) {
+    billingDatabases <- tibble::tibble(
+      databaseId = unlist(lapply(billingDatabases, function(x) {x$databaseId})),
+      label = unlist(lapply(billingDatabases, function(x) {x$label})),
+      description = unlist(lapply(billingDatabases, function(x) { if(nzchar(x$description)) x$description else NA_character_ })),
+      owner = lapply(billingDatabases, function(x) {x$owner}),
+      formCount = unlist(lapply(billingDatabases, function(x) {x$formCount})),
+      userCount = unlist(lapply(billingDatabases, function(x) {x$userCount})),
+      basicUserCount = unlist(lapply(billingDatabases, function(x) {x$basicUserCount})),
+      recordCount = unlist(lapply(billingDatabases, function(x) {x$recordCount})),
+      lastRecordUpdate = unlist(lapply(billingDatabases, function(x) {x$lastRecordUpdate})),
+      billingAccountId = unlist(lapply(billingDatabases, function(x) {x$billingAccountId})),
+      suspended = unlist(lapply(billingDatabases, function(x) {x$suspended})),
+      publishedTemplate = unlist(lapply(billingDatabases, function(x) {x$publishedTemplate}))
+    )
+    return(billingDatabases)
+  } else {
+    return(billingDatabases)
+  }
+}
+
+
+#' @title getBillingDomains
+#' @description Billing account email domain info
+#' @param userId User ID
+#' @return Information on billing account email domain in list output
+#' @rdname getBillingDomains
+#' @export 
+
+getBillingDomains <- function(userId) {
+  billingDomains <- getResource(paste0("/billingAccounts/", userId, "/domains"), task = "Getting billing account domains")
+  return(billingDomains)
+}
+
+
+#' @title getBillingUsers
+#' @description Billing account users
+#' @param asDataFrame Output as data.frame, Default: TRUE
+#' @param userId User ID
+#' @return Billing account user(s) in list or data.frame output
+#' @rdname getBillingUsers
+#' @export 
+#' @importFrom tibble tibble
+
+getBillingUsers <- function(asDataFrame = TRUE, userId) {
```

Comment:
FYI: this endpoint also accepts an optional `databaseId` query parameter that filters to just the users who have access to a given database. As an account owner, you have permission to access this list, even if you don't have access to the database itself.

Reply:
(I realize now that query parameters are not listed in the online API documentation, I will try to fix this soon)

Reply:
right yeah, that's why i have `getBillingAccountUsers()` and `getBillingAccountDatabaseUsers()`. the latter uses the query parameter `/users?databaseId=` to filter for a specific database